JavaScript/packages/recognizers-date-time/src/resources/englishDateTime.ts

```diff
@@ -8,8 +8,8 @@
 
 import { BaseDateTime } from "./baseDateTime";
 export namespace EnglishDateTime {
-	export const TillRegex = `(?<till>to|till|til|until|thru|through|--|-|—|——|~)`;
-	export const RangeConnectorRegex = `(?<and>and|through|to|--|-|—|——)`;
+	export const TillRegex = `(?<till>to|till|til|until|thru|through|--|-|—|——|~|–)`;
```

Comment:
This is an 'n-dash', a common char for ranges. Can you test if using \u2013 is enough to make the JS port work?
The Boolean recognizers had a similar issue with some unicode chars..

Reply:
BTW, we should probably add it to the equivalent number range case.